pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -620,11 +621,18 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
       Map<ServerInstance, Pair<List<String>, List<String>>> realtimeRoutingTable = null;
       List<String> unavailableSegments = new ArrayList<>();
       int numPrunedSegmentsTotal = 0;
+      boolean offlineTableDisabled = _routingManager.isTableDisabled(
```

Comment:
Move this check into `if (offlineBrokerRequest != null)`, and use `offlineTableName`. When table is disabled, no need to compute the routing table.
Same for real-time table handling

Reply:
Gotcha. I've made that fix, so it only computes the routing table if the table is enabled.